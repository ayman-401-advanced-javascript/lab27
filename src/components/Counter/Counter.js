import React from 'react';


class Counter extends React.Component {
    constructor(properties) {
      super(properties)
      this.state = {
        counter: 0
      };
    }
   
    handLeIncrementClick = e => {
      e.preventDefault();
      let counte = this.state.counter++;
      this.setState({counte})
  }
  handLeDecrementClick = e => {
    e.preventDefault();
    let counte = this.state.counter--;
    this.setState({counte})
  }
  
  render() {
    return (
      <section>
        <h3>{this.state.counter}</h3>
        <button className="Increment button" onClick={this.handLeIncrementClick}>click for Increment</button>
        <button className="Decrement button" onClick={this.handLeDecrementClick}>click for Decrement</button>
      </section>
    )
  }
  }

export default Counter;