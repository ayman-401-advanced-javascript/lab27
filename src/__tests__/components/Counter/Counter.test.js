import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Counter from '../../../components/Counter/Counter.js';

describe('<Counter />', ()=>{
    it('exists at the start of our application',()=>{
        let app = shallow(<Counter />);
        expect(app.find('h3').exists()).toBeTruthy();
    });
    it('Increment the state on click',()=>{
        let app = mount(<Counter />);
        let button = app.find('.Increment');
        expect(app.state('counter')).toBe(0)
        button.simulate('click');
        expect(app.state('counter')).toBe(1)
    });
    it('Decrement the state on click',()=>{
        let app = mount(<Counter />);
        let button = app.find('.Decrement');
        expect(app.state('counter')).toBe(0)
        button.simulate('click');
        expect(app.state('counter')).toBe(-1)
    });
    it('renders correctly', () => {
        const tree = renderer.create(<Counter />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});