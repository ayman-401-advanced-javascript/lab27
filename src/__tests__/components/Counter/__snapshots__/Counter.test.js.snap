// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<section>
  <h3>
    0
  </h3>
  <button
    className="Increment button"
    onClick={[Function]}
  >
    click for Increment
  </button>
  <button
    className="Decrement button"
    onClick={[Function]}
  >
    click for Decrement
  </button>
</section>
`;
